### PR TITLE
Poll refactor

### PR DESCRIPTION
**Pull request by benghaziboy**
@collin @mparent61 @MelizzaP @MichaelButkovic 

Moved things around as best I could for the time being. Big problem is the relational aspect of all of this, so poke holes where necessary. I'm "done" with the starwipe portion of this, but I'll open that up once we've properly dissected this part.

### CODE REVIEW

File: bulbs/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "0.8.5"
+__version__ = "0.8.6"
```
Comment (MichaelButkovic) [inline bulbs/__init__.py]: i just took `0.8.6` with the recirc stuff

**Comment by MichaelButkovic** [comment]
:+1:

**Comment by mparent61** [comment]
:+1:

File: bulbs/poll/sodahead.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    def __init__(self, detail):
+        self.detail = detail
+
+
+class SodaheadResponseFailure(APIException):
+    status_code = 400
+    default_detail = "Error from third-party poll provider"
+
+    def __init__(self, detail):
+        self.detail = detail
```
Comment (MelizzaP) [inline bulbs/poll/sodahead.py]: Is this duplicated from bulbs/poll/mixins.py intentionally?

Comment (benghaziboy) [inline bulbs/poll/sodahead.py]: ah nope.

**Comment by MelizzaP** [comment]
:+1:  other than my q

**Comment by collin** [comment]
Looks okay.

**Comment by benghaziboy** [comment]
so do you pal

**Pull request merged**